### PR TITLE
Context closing during snapshot bugfix

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Reasoning behind the pull request
- activeDB marker was set even though there was a context closing err during snapshot
  
## Proposed changes
- return err during snapshot if the db is closing

## Testing procedure
- normal system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by iulianpascalau — COMMENTED** [review]
forgotten tests from this commit? https://github.com/multiversx/mx-chain-go/pull/5743/commits/cf589d491f98c19a2dbee3cee19efa0c8cdcfd56

**Comment by BeniaminDrasovean** [comment]
No, those tests were related to the changes you made.

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: trie/node.go [hunk 1]
```diff
@@ -275,14 +276,22 @@ func shouldStopIfContextDoneBlockingIfBusy(ctx context.Context, idleProvider Idl
 	}
 }
 
-func treatCommitSnapshotError(err error, hash []byte, missingNodesChan chan []byte) {
+func treatCommitSnapshotError(err error, hash []byte, missingNodesChan chan []byte) (bool, error) {
```
Comment (AdoAdoAdo) [inline trie/node.go:279]: maybe use named return values? 
not very clear what the bool return value signifies.

File: trie/node.go [hunk 2]
```diff
@@ -275,14 +276,22 @@ func shouldStopIfContextDoneBlockingIfBusy(ctx context.Context, idleProvider Idl
...
+	if err == nil {
+		return false, nil
+	}
+
+	if !strings.Contains(err.Error(), core.GetNodeFromDBErrorString) {
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply trie/node.go:284]: Done

File: trie/node.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply trie/node.go:279]: Done.

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.5.15-dev-config-e134e5314d -> context-closing-during-sna-1a384abaf7

--- Specific errors ---

block hash does not match 11752
wrong nonce in block 4843
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  3
Nr. of all WARNS:  2758
Nr. of new ERRORS:  3
Nr. of new WARNS:  29
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

**Pull request merged**